packages/buidler-core/src/internal/core/providers/http.ts

```diff
@@ -85,11 +85,20 @@ export class HttpProvider extends EventEmitter {
         }
       });
 
-      const json = await response.json();
+      const text = await response.text();
+      let json: any;
+      try {
+        json = JSON.parse(text);
+      } catch (error) {
+        throw new FetchError(
```

Comment:
Thanks for fixing this!

I think it'd be better to also throw `INVALID_JSON_RESPONSE` in this case, don't you?

Reply:
Makes sense, the FetchError was implemented to be backwards compatible with having only response.json() there. If you look at the implemntation of .json(), that's what you will see. 😄 

I will switch it to a Buidler specific error now.